### PR TITLE
BUG: fix error in Poisson when offset or exposure is a pd.Series

### LINKED ISSUE #7015: Error when trying to use an exposure term or offset term with `statsmodels.discrete.count_model.Poisson`
#### Describe the bug

Error when trying to use an exposure term or offset term with `statsmodels.discrete.count_model.Poisson` and `statsmodels.discrete.count_model.ZeroInflatedPoisson`.

```python
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from patsy import dmatrices

data = pd.read_csv('https://gist.githubusercontent.com/sachinsdate/09cfd42b7701c48ec68b04c786786434/raw/4b50e718a83a4b20adcff5552ccea8f1054ce919/fish.csv')

Pmodel1 = smf.glm('FISH_COUNT ~ C(CAMPER)', data=data, family=sm.families.Poisson(), exposure=data['PERSONS']).fit() # this works
Pmodel1.summary()


y_mat, X_mat = dmatrices('FISH_COUNT ~ C(CAMPER)', data, return_type='dataframe')

Pmodel2 = sm.Poisson(endog=y_mat, exog=X_mat, exposure=data['PERSONS']).fit() # this gives an error

ZIPmodel = sm.ZeroInflatedPoisson(endog=y_mat, exog=X_mat, exposure=data['PERSONS'], exog_infl=X_mat, inflation='logit').fit() # this gives an error
```
<details><summary>Traceback</summary>

```python
ValueError                                Traceback (most recent call last)
<ipython-input-8-623bd13d22ad> in <module>
----> 1 Pmodel2 = sm.Poisson(endog=y_mat, exog=X_mat, exposure=data['PERSONS']).fit() # this gives an error

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/discrete/discrete_model.py in fit(self, start_params, method, maxiter, full_output, disp, callback, **kwargs)
   1104                                              disp=disp,
   1105                                              callback=callback,
-> 1106                                              **kwargs)
   1107 
   1108         if 'cov_type' in kwargs:

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/discrete/discrete_model.py in fit(self, start_params, method, maxiter, full_output, disp, callback, **kwargs)
    231                              disp=disp,
    232                              callback=callback,
--> 233                              **kwargs)
    234 
    235         return mlefit  # It is up to subclasses to wrap results

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/base/model.py in fit(self, start_params, method, maxiter, full_output, disp, fargs, callback, retall, skip_hessian, **kwargs)
    525                                                        callback=callback,
    526                                                        retall=retall,
--> 527                                                        full_output=full_output)
    528 
    529         # NOTE: this is for fit_regularized and should be generalized

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/base/optimizer.py in _fit(self, objective, gradient, start_params, fargs, kwargs, hessian, method, maxiter, full_output, disp, callback, retall)
    216                             disp=disp, maxiter=maxiter, callback=callback,
    217                             retall=retall, full_output=full_output,
--> 218                             hess=hessian)
    219 
    220         optim_settings = {'optimizer': method, 'start_params': start_params,

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/base/optimizer.py in _fit_newton(f, score, start_params, fargs, kwargs, disp, maxiter, callback, retall, full_output, hess, ridge_factor)
    314     while (iterations < maxiter and np.any(np.abs(newparams -
    315             oldparams) > tol)):
--> 316         H = np.asarray(hess(newparams))
    317         # regularize Hessian, not clear what ridge factor should be
    318         # keyword option with absolute default 1e-10, see #1847

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/base/model.py in hess(params, *args)
    507 
    508             def hess(params, *args):
--> 509                 return self.hessian(params, *args) / nobs
    510         else:
    511             def score(params, *args):

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels/discrete/discrete_model.py in hessian(self, params)
   1317         X = self.exog
   1318         L = np.exp(np.dot(X,params) + exposure + offset)
-> 1319         return -np.dot(L*X.T, X)
   1320 
   1321     def hessian_factor(self, params):

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/pandas/core/ops/common.py in new_method(self, other)
     63         other = item_from_zerodim(other)
     64 
---> 65         return method(self, other)
     66 
     67     return new_method

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/pandas/core/ops/__init__.py in wrapper(left, right)
    343         result = arithmetic_op(lvalues, rvalues, op)
    344 
--> 345         return left._construct_result(result, name=res_name)
    346 
    347     wrapper.__name__ = op_name

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/pandas/core/series.py in _construct_result(self, result, name)
   2755         # We do not pass dtype to ensure that the Series constructor
   2756         #  does inference in the case where `result` has object-dtype.
-> 2757         out = self._constructor(result, index=self.index)
   2758         out = out.__finalize__(self)
   2759 

~/miniconda3/envs/stdsci/lib/python3.6/site-packages/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    312                     if len(index) != len(data):
    313                         raise ValueError(
--> 314                             f"Length of passed values is {len(data)}, "
    315                             f"index implies {len(index)}."
    316                         )

ValueError: Length of passed values is 2, index implies 250.
```
</details>

Fitting a Poisson model with an exposure or offset term works if done through the formula API but trying to do so using the non-fomrula API gives the error above. This is especialyl a problem when the model is not accessible with the formula API e.g. `statsmodels.discrete.count_model.ZeroInflatedPoisson`.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.6.7.final.0
OS: Linux 5.8.0-1-default #1 SMP Tue Aug 4 07:30:59 UTC 2020 (9bc0044) x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8

statsmodels
===========

Installed: 0.12.0 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/statsmodels)

Required Dependencies
=====================

cython: Not installed
numpy: 1.19.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/numpy)
scipy: 1.5.2 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/scipy)
pandas: 1.1.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/pandas)
    dateutil: 2.8.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/dateutil)
patsy: 0.5.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.3.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/matplotlib)
    backend: module://ipykernel.pylab.backend_inline 
cvxopt: Not installed
joblib: Not installed

Developer Tools
================

IPython: 7.11.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/IPython)
    jinja2: Not installed
sphinx: Not installed
    pygments: 2.6.1 (/home/ilya/miniconda3/envs/stdsci/lib/python3.6/site-packages/pygments)
pytest: Not installed
virtualenv: Not installed

</details>

### PR DESCRIPTION
**Pull request by esmucler**
- [X] closes #7015
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>
Just do np.asarray in the __init__ method of CountModel. Added a test that fails on master but passes here

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/discrete/tests/test_count_model.py [hunk 1]
```diff
@@ -643,3 +643,17 @@ def test_zero_nonzero_mean(self):
...
+    def test_pd_offset_exposure(self):
+        endog = pd.DataFrame({'F': [0.0, 0.0, 0.0, 0.0, 1.0]})
+        exog = pd.DataFrame({'I': [1.0, 1.0, 1.0, 1.0, 1.0], 'C': [0.0, 1.0, 0.0, 1.0, 0.0]})
+        exposure = pd.Series([1, 1, 1, 2, 1])
+        offset = pd.Series([1, 1, 1, 2, 1])
+        sm.Poisson(endog=endog, exog=exog, offset=offset).fit()
+        inflations = ['logit', 'probit']
+        for inflation in inflations:
+            sm.ZeroInflatedPoisson(endog=endog, exog=exog, exposure=exposure, exog_infl=exog,
+                                   inflation=inflation).fit()
```
Comment (josef-pkt) [inline statsmodels/discrete/tests/test_count_model.py:659]: missing one newline at end of file

**Comment by josef-pkt** [comment]
Thanks, looks good to me

one pep-8 issue in formatting, missing newline

File: statsmodels/discrete/tests/test_count_model.py (hunk 1 above)
Comment (esmucler) [reply statsmodels/discrete/tests/test_count_model.py]: this is fixed now

**Comment by esmucler** [comment]
> Thanks, looks good to me
> 
> one pep-8 issue in formatting, missing newline

The pep-8 issue is fixed, do you think this is good to go?

**Comment by josef-pkt** [comment]
I cleaned this up a bit in PR #7228

Sorry for the delay, and thanks for the PR

**Pull request closed without merging**